### ISSUE TITLE
problem with -- list index out of range

### ISSUE THREAD

**Issue by sunhaifeng123** [ZhengXia/dapars#11]
Hi : 
I'm using Dapars to find APA,but some problems seem to fused me : 
python /home/pc/biosoft/dapars/src/DaPars_Extract_Anno.py -b $refbed -s $genesymbol -o Dapars_extracted_3UTR.bed

Generating regions ...
Traceback (most recent call last):
  File "/home/pc/biosoft/dapars/src/DaPars_Extract_Anno.py", line 151, in <module>
    Extract_Anno_main(sys.argv[1:])
  File "/home/pc/biosoft/dapars/src/DaPars_Extract_Anno.py", line 139, in Extract_Anno_main
    Annotation_prepar_3UTR_extraction(gene_bed_file, gene_symbol_annotation_file,output_extract_file)
  File "/home/pc/biosoft/dapars/src/DaPars_Extract_Anno.py", line 17, in Annotation_prepar_3UTR_extraction
    gene_symbol = fields[1]
IndexError: list index out of range

do u know what's happened ? how should i solve this problem?
Thanks very much !

Haifeng Sun
China Nanjing Medical University

**Comment by obenno** [comment]
Hi Sun,

Is your gene symbol file properly formatted? Could you provide some example
lines?

Bests,
Zhixia

**Comment by Riverhuang42** [comment]
Hi,

I had some problems when I am running DaPars_main.py.
my  extracted_3UTR.bed is like
chr5	100589900	100592136	ENSMUST00000239512.1|Lin54|chr5|-	0	-
chr17	37314356	37314706	ENSMUST00000174016.8|Zfp57|chr17|+	0	+
chr15	79173371	79173485	ENSMUST00000174375.2|Pla2g6|chr15|-	0	-

wig file is like
chr1    0       3191839 0
chr1    3191839 3191841 1
chr1    3191841 3191929 2
chr1    3191929 3192634 0

Can anyone give some suggestions?
Thanks

**Comment by gadiscymraes** [comment]
I also have the same error message when I am running DaPars_main.py

Did anyone find a solution?

DaPars_Extract_Anno.py ran fine for me.
My files shown below.

$ head hg38_2_8_2022_Refseq_id_from_UCSC.txt
#name   name2
NM_000299       PKP1
NM_001276351    C1orf141
NM_001005337    PKP1
NM_001276352    C1orf141
NR_075077       C1orf141
NM_001042681    RERE
NM_012102       RERE


$ head hg38_refseq_extracted_3UTR.bed
chr15   41502963        41503551        NM_002220|ITPKA|chr15|+ 0       +
chr7    151238973       151239156       NM_001003802|SMARCD3|chr7|-     0       -
chr9    92711363        92715463        NM_001003800|BICD2|chr9|-       0       -
chr19   50417842        50418018        NM_001308632|POLD1|chr19|+      0       +
chr3    112815711       112815975       NM_001008784|CD200R1L|chr3|-    0       -
chr10   124757834       124760503       NM_001304467|EEF1AKMT2|chr10|-  0       -
chr12   51891186        51891727        NM_001304460|ANKRD33|chr12|+    0       +
chr11   92890491        92896473        NM_001008781|FAT3|chr11|+       0       +


$ head hg38_refseq_whole_gene.bed
chr1    201283451       201332993       NM_000299       0       +       201283702       201328836       0    8526,39561,40976,41489,42302,45310,46622,
chr1    67092165        67134970        NM_001276351    0       -       67093004        67127240        0
chr1    201283505       201332989       NM_001005337    0       +       201283702       201328836       0    9507,40922,41435,42248,45256,46568,
chr1    67092165        67134970        NM_001276352    0       -       67093579        67127240        0
chr1    67092165        67134970        NR_075077       0       -       67134970        67134970        0
chr1    8352403 8817640 NM_001042681    0       -       8355086 8656297 0       23      2717,181,147,721,223,2342,13408,70323,113521,142659,145001,156223,188810,204071,205014,262157,271906,303569,464756,
chr1    8352403 8817640 NM_012102       0       -       8355086 8656297 0       24      2717,181,147,721,223,2342,13408,70323,113521,142659,145001,156223,188810,204071,205014,262157,271906,303569,440000,464756,
chr1    41847188        42035934        NR_038261       0       -       42035934        42035934        0
chr1    8352403 8423832 NM_001042682    0       -       8355086 8364133 0       13      2717,181,147,721,223,
chr1    33513997        34165230        NM_001281956    0       -       33519517        34165097        0    178,189,210,117,70,119,105,97,125,204,96,114,117,195,188,139,192,203,127,192,157,170,189,216,117,189,188,139,27,27132,28722,32039,36179,43736,45302,53595,55376,57534,58508,63298,66755,69644,72506,73090,86867,88371,9128184755,186519,195091,200589,202288,210199,210518,211351,212549,225142,229282,258571,274602,278425,296745,3057

**Comment by MTDeng** [comment]
I had the same problem when I am running DaPars2_Multi_Sample_Multi_Chr.py
Traceback (most recent call last):
  File "/gss1/home/dmt20191202/conda/biosoft/dapars2/src/DaPars2_Multi_Sample_Multi_Chr.py", line 449, in <module>
    De_Novo_3UTR_Identification_Loading_Target_Wig_for_TCGA_Multiple_Samples_Multiple_threads_Main3_shared_list(sys.argv)
  File "/gss1/home/dmt20191202/conda/biosoft/dapars2/src/DaPars2_Multi_Sample_Multi_Chr.py", line 110, in De_Novo_3UTR_Identification_Loading_Target_Wig_for_TCGA_Multiple_Samples_Multiple_threads_Main3_shared_list
    fh=open(sys.argv[2],'r')
IndexError: list index out of range

**Comment by heiyuemenyue** [comment]
I've also met the same problem:

        {```$ python3 DaPars_Extract_Anno.py -b mm10_refseq_whole_gene.bed -s mm10_Refseq_id_from_UCSC.txt -o mm10_refseq_extracted_3UTR.bed
        Generating regions ...
        Traceback (most recent call last):
          File "DaPars_Extract_Anno.py", line 151, in <module>
            Extract_Anno_main(sys.argv[1:])
          File "DaPars_Extract_Anno.py", line 139, in Extract_Anno_main
            Annotation_prepar_3UTR_extraction(gene_bed_file, gene_symbol_annotation_file,output_extract_file)
          File "DaPars_Extract_Anno.py", line 17, in Annotation_prepar_3UTR_extraction
            gene_symbol = fields[1]
        IndexError: list index out of range```}

After trying tail mm10_Refseq_id_from_UCSC.txt I found some unexpected text at the end of the file:

```
NM_001378759.1  LOC100861691
XM_017318925.1  LOC100861691
---------------------------------------------------------------------------
Can't start query:
UPDATE userDb SET contents='jsh_pageVertPos=0&hgta_fs.check.mm10.ncbiRefSeq.txEnd=0&hgta_fs.linked.mm10.ncbiRefSeqSelect=0&sendToGreat=0&hgta_fs.check.mm10.ncbiRefSeq.exonStarts=0&clade=mammal&hgta_database=mm10&hgta_fs.linked.mm10.ncbiRefSeqCds=0&hgta_group=genes&hgta_ctDesc=table+browser+query+on+ncbiRefSeq&hgta_fs.check.mm10.ncbiRefSeq.strand=0&sendToGalaxy=0&hgta_fs.linked.mm10.ncbiRefSeqCurated=0&hgta_fs.check.mm10.ncbiRefSeq.chrom=0&hgta_printCustomTrackHeaders=0&trackHubs=&hgta_fs.check.mm10.ncbiRefSeq.name=1&hgta_outSep=tab&hgta_fs.check.mm10.ncbiRefSeq.txStart=0&hgta_fieldSelectTable=mm10.ncbiRefSeq&hgta_ctName=tb_ncbiRefSeq&hgta_fs.check.mm10.ncbiRefSeq.exonCount=0&hgta_regionType=genome&hgta_fs.check.mm10.ncbiRefSeq.score=0&org=Mouse&textSize=12&hgta_table=ncbiRefSeq&hgta_compressType=none&hgta_track=refSeqComposite&hgta_fs.linked.mm10.ncbiRefSeqPredicted=0&fbQual=whole&hgta_ctVis=pack&hgta_ctUrl=&hgta_outFileName=mm10_Refseq_id_from_UCSC.txt&fbIntronBases=0&db=mm10&textFont=Helvetica&hgta_fs.linked.mm10.ncbiRefSeqPsl=0&hgta_fs.check.mm10.ncbiRefSeq.exonFrames=0&hgta_fs.check.mm10.ncbiRefSeq.cdsStart=0&position=chr12%3A56%2C694%2C976%2D56%2C714%2C605&hgta_fs.check.mm10.ncbiRefSeq.cdsEnd=0&hgta_fs.check.mm10.ncbiRefSeq.bin=0&hgta_outputType=selectedFields&hgta_fs.linked.mm10.ncbiRefSeqLink=0&fbDownBases=200&hgta_fs.linked.mm10.seqNcbiRefSeq=0&hgta_fs.check.mm10.ncbiRefSeq.cdsEndStat=0&hgta_fs.check.mm10.ncbiRefSeq.cdsStartStat=0&textStyle=Normal&fbUpBases=200&fbExonBases=0&hgta_fs.check.mm10.ncbiRefSeq.name2=1&hgta_fs.check.mm10.ncbiRefSeq.exonEnds=0',lastUse=now(),useCount=30  where id=2691270464 and sessionKey='Rl0UNgyN8k4rymNn2af0syoHdzQD'

---------------------------------------------------------------------------
---------------------------------------------------------------------------
mySQL error 2006: Server has gone away (profile=<noProfile>, host=hgnfs1, db=hgcentral)
---------------------------------------------------------------------------
```

in the DaPars_Extract_Anno.py line 15 to 17 it coded:

```
 if num_line > 0:
            fields = line.strip('\n').strip('\r').split('\t')
            gene_symbol = fields[1]
```

which tried to split the mm10_Refseq_id_from_UCSC.txt into to columns and then catch the first column, but the unexpected text at the end of the file ruined the program, which may cause the problem. Then I tried to delete those text and the the problem seemed to be solved.

**_In summary: check whether there have the similar text at the end of your mm10_Refseq_id_from_UCSC.txt or  mm10_refseq_whole_gene.bed, if there has, delete them and try again, those text may caused by the sever error of Genome so I can't confirm whether those file are complete._**

Hope my comment helps.
